src/main/java/uk/ac/ebi/ampt2d/metadata/persistence/entities/WebResource.java

```diff
@@ -61,6 +63,8 @@
     @URL
     @JsonProperty
     @Column(nullable = false, columnDefinition = "TEXT")
+    @NotEmpty
+    @Pattern(message = "Must be a valid URL.", regexp="^(https?|ftp|file):(//|\\\\)[-a-zA-Z0-9+&@#/%?=~_|!:,.;]*[-a-zA-Z0-9+&@#/%=~_|]")
```

Comment:
Sure, i have removed "file" protocol and its gets blocked also. But existing @URL code bars no-protocol URLs. If we remove @URL we will loose JSR validations for URL.

Reply:
Ah, I forgot there is a `@URL` annotation. It seems that `@URL` validation can take a few optional settings, like `protocol` and `regexp`: https://docs.jboss.org/hibernate/validator/5.4/api/org/hibernate/validator/constraints/URL.html

If these work for our purpose, I would suggest to use these instead of the `@Pattern` annotation.